holoviews/core/data/interface.py

```diff
@@ -50,22 +50,31 @@ def initialize(cls, eltype, data, kdims, vdims, datatype=None):
             kdims = pvals.get('kdims') if kdims is None else kdims
             vdims = pvals.get('vdims') if vdims is None else vdims
 
+        if datatype is None:
+            datatype = eltype.datatype
+
         # Process Element data
         if isinstance(data, NdElement):
             kdims = [kdim for kdim in kdims if kdim != 'Index']
-        elif hasattr(data, 'interface') and issubclass(data.interface, Interface):
-            data = data.data
+        elif (hasattr(data, 'interface') and issubclass(data.interface, Interface)):
```

Comment:
No, that's waiting on us supporting bin edges in the interfaces somehow, at which point both Histogram and QuadMesh will be supported. Iris has a concept for them and I don't think it would be to hard to add to the dict interface, I don't know about xarray.